axelrod/tests/strategies/test_gobymajority.py

```diff
@@ -24,40 +25,55 @@ class TestHardGoByMajority(TestPlayer):
     }
 
     def test_strategy(self):
-        # Starts by defecting.
         self.first_play_test(self.eq_play)
-        # If opponent cooperates at least as often as they defect then the
-        # player defects.
-        self.responses_test([self.eq_play], [C, D, D, D], [D, D, C, C])
-        # If opponent defects strictly more often than they defect then the
-        # player defects.
-        self.responses_test([D], [C, C, D, D, C], [D, D, C, C, D])
-        # If opponent cooperates strictly more often than they defect then the
-        # player cooperates.
-        self.responses_test([C], [C, C, D, D, C], [D, C, C, C, D])
-
-    def test_default_soft(self):
+
+        expected, opponent_actions = self.get_infinite_memory_depth_actions()
+        self.versus_test(MockPlayer(actions=opponent_actions),
+                         expected_actions=expected)
+
+    def get_infinite_memory_depth_actions(self):
+        opponent_actions = [C, D, D]
+        first_three = [(self.eq_play, C), (C, D), (self.eq_play, D)]
+        second_three = [(D, C), (self.eq_play, D), (D, D)]
+        subsequent = [(D, C), (D, D), (D, D)]
+        expected = first_three + second_three + subsequent * 10
+        return expected, opponent_actions
+
+    def test_memory_depth(self):
+        memory_depth = 4
+        opponent_actions = [C, C, C, D, D, D]
+        first_six = [(self.eq_play, C), (C, C), (C, C),
+                     (C, D), (C, D), (self.eq_play, D)]
+        subsequent = [(D, C), (D, C), (self.eq_play, C),
+                      (C, D), (C, D), (self.eq_play, D)]
+
+        expected = first_six + subsequent * 10
+        self.versus_test(MockPlayer(actions=opponent_actions),
+                         expected_actions=expected,
+                         init_kwargs={'memory_depth': memory_depth})
+
+    def test_soft_value(self):
         player = self.player()
-        self.assertEqual(player.soft, self.default_soft)
+        self.assertFalse(player.soft)
 
 
 class TestGoByMajority(TestHardGoByMajority):
 
     name = "Soft Go By Majority"
     player = axelrod.GoByMajority
-    default_soft = True
     eq_play = C
-
-    def test_strategy(self):
-        # In case of equality (including first play), cooperates.
-        super().test_strategy()
-
-        # Test tie break rule for soft=False
-        player = self.player(soft=False)
-        opponent = axelrod.Cooperator()
-        self.assertEqual('D', player.strategy(opponent))
-
-    def test_soft(self):
+    soft = True
+
+    def test_set_soft_to_false(self):
+        self.eq_play = D
+        expected, opponent_actions = self.get_infinite_memory_depth_actions()
+        self.versus_test(MockPlayer(actions=opponent_actions),
+                         expected_actions=expected, init_kwargs={'soft': False})
+        self.eq_play = C
```

Comment:
Not sure why this needs to be a class attribute. (I'm not actually sure what this is doing.)